sky/packages/sky/lib/src/animation/animation_performance.dart

```diff
@@ -31,12 +50,22 @@ enum AnimationStatus {
 /// may also take direct control of the timeline by manipulating [progress], or
 /// [fling] the timeline causing a physics-based simulation to take over the
 /// progression.
-class AnimationPerformance {
+class AnimationPerformance implements WatchableAnimationPerformance {
   AnimationPerformance({ AnimatedVariable variable, this.duration }) :
     _variable = variable {
     _timeline = new Timeline(_tick);
   }
 
+  /// Returns an [AnimationPerformanceView] that wraps this performance,
+  /// so that a pointer to this object can be passed around without
+  /// allowing users of that pointer to mutate the AnimationPerformance state.
+  AnimationPerformanceView _view;
+  AnimationPerformanceView get view {
```

Comment:
Could this simply return `this as WatchableAnimationPerformance`? Then have TransitionBase accept that interface. The user wouldn't be able to modify the performance without casting.